--- OpenBLAS/cblas.h.old	2025-07-01 16:41:19.256253903 +0800
+++ OpenBLAS/cblas.h	2025-07-01 16:42:43.140860955 +0800
@@ -31,13 +31,6 @@
 typedef void (*openblas_threads_callback)(int sync, openblas_dojob_callback dojob, int numjobs, size_t jobdata_elsize, void *jobdata, int dojob_data);
 void openblas_set_threads_callback_function(openblas_threads_callback callback);
 
-#ifdef OPENBLAS_OS_LINUX
-/* Sets thread affinity for OpenBLAS threads. `thread_idx` is in [0, openblas_get_num_threads()-1]. */
-int openblas_setaffinity(int thread_idx, size_t cpusetsize, cpu_set_t* cpu_set);
-/* Queries thread affinity for OpenBLAS threads. `thread_idx` is in [0, openblas_get_num_threads()-1]. */
-int openblas_getaffinity(int thread_idx, size_t cpusetsize, cpu_set_t* cpu_set);
-#endif
-
 /* Get the parallelization type which is used by OpenBLAS */
 int openblas_get_parallel(void);
 /* OpenBLAS is compiled for sequential use  */
