--- OpenBLAS/driver/others/blas_server.c.old	2025-07-01 16:41:14.462277959 +0800
+++ OpenBLAS/driver/others/blas_server.c	2025-07-01 16:42:03.288041251 +0800
@@ -343,39 +343,6 @@
 
 static int increased_threads = 0;
 
-#ifdef OS_LINUX
-extern int openblas_get_num_threads(void);  
-
-int openblas_setaffinity(int thread_idx, size_t cpusetsize, cpu_set_t* cpu_set) {
-  const int active_threads = openblas_get_num_threads();
-
-  if (thread_idx < 0 || thread_idx >= active_threads) {
-    errno = EINVAL;
-    return -1;
-  }
-
-  pthread_t thread = (thread_idx == active_threads - 1)
-      ? pthread_self()
-      : blas_threads[thread_idx];
-
-  return pthread_setaffinity_np(thread, cpusetsize, cpu_set);
-}
-int openblas_getaffinity(int thread_idx, size_t cpusetsize, cpu_set_t* cpu_set) {
-  const int active_threads = openblas_get_num_threads();
-
-  if (thread_idx < 0 || thread_idx >= active_threads) {
-    errno = EINVAL;
-    return -1;
-  }
-
-  pthread_t thread = (thread_idx == active_threads - 1)
-      ? pthread_self()
-      : blas_threads[thread_idx];
-
-  return pthread_getaffinity_np(thread, cpusetsize, cpu_set);
-}
-#endif
-
 static void* blas_thread_server(void *arg){
 
   /* Thread identifier */
